tests/test_domain_c.py

```diff
@@ -426,7 +426,7 @@ def test_union_definitions():
     check('struct', '{key}A', {1: 'A'})
 
 
-def test_union_definitions():
+def test_union_definitions2():
```

Comment:
@jakobandersen This case overrides the above one. Could you give a good name for this in another PR, please?

Reply:
This one should keep the name ``test_union_definitions`` while the above should be renamed to ``test_struct_definitions``.

Reply:
Done in https://github.com/sphinx-doc/sphinx/pull/8433.